src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -352,14 +351,13 @@ private UnboundLambda BindAnonymousFunction(AnonymousFunctionExpressionSyntax sy
 
             var lambda = AnalyzeAnonymousFunction(syntax, diagnostics);
             var data = lambda.Data;
-            if (data.HasExplicitlyTypedParameterList)
+
+            int firstDefault = -1;
+            for (int i = 0; i < lambda.ParameterCount; i++)
             {
-                int firstDefault = -1;
-                for (int i = 0; i < lambda.ParameterCount; i++)
+                var paramSyntax = lambda.ParameterSyntax(i);
+                if (paramSyntax is not null)
```

Comment:
Why can `paramSyntax` be `null`? If the lambda has non-zero `ParameterCount` I would expect it to have corrsponding parameter syntaxes.

Reply:
It can be null in some tests for parameters with modifiers without parameter parentheses (e.g. `ref x => x`), which is illegal per the spec. Without the underlying changes, now permitting the `null` parameter syntax, the assertion would fail.

Since the modifiers are not allowed in single-parameter unparenthesized lambda expressions, we need to at least expect the case to arise.

Reply:
i don't really get it either.  i would not expect any sort of lambda to have a nulll parameter syntax.

Reply:
> Since the modifiers are not allowed in single-parameter unparenthesized lambda expressions, we need to at least expect the case to arise.

this sentence doesn't make sense.  Just because the modifiers are not legal doesn't mean you get a null parameter syntax.

Reply:
Made changes to preserve this, but kept the removal of the check about having an explicitly typed parameter list. To circumvent breaks, lambda binding data for simple lambda expressions (parameters without parentheses) now contain a singleton separated syntax list of parameter syntaxes.